odl/solvers/functional/default_functionals.py

```diff
@@ -453,18 +453,21 @@ class IndicatorLpUnitBall(Functional):
     norm.
     """
 
-    def __init__(self, space, exponent):
+    def __init__(self, domain, exponent, range=None):
         """Initialize a new instance.
 
         Parameters
         ----------
-        space : `DiscreteLp` or `TensorSpace`
+        domain : `DiscreteLp` or `TensorSpace`
             Domain of the functional.
         exponent : int or infinity
             Specifies wich norm to use.
         """
-        super(IndicatorLpUnitBall, self).__init__(space=space, linear=False)
-        self.__norm = LpNorm(space, exponent)
+        super(IndicatorLpUnitBall, self).__init__(domain=domain, linear=False,
+                                                  range=range)
+        # HELP: The LpNorm should always have a range in the real numbers so
+        #       that it is comparable to 1, right?
```

Comment:
Yeah, so you can override the `Functional` default before the `super` call:
```python
if range is None:
    range = RealNumbers()
```
And then see what happens :laughing:

Reply:
Isn't this more like what we want to have: `self.__norm = LpNorm(domain, exponent, range=RealNumbers())`
Then the range of the class itself still can be complex if needed

Reply:
Oh I thought you were in `LpNorm`, didn't realize it was the indicator. There, yes, I agree with your suggestion.